### ISSUE TITLE
not work "builtinGlobals" in rule "no-redeclare" 

### ISSUE THREAD

**Issue by boggddan** [eslint/eslint#8814]
**Tell us about your environment**

* **ESLint Version:**
v4.0.0
* **Node Version:**
v8.0.0
* **npm Version:**
v5.0.0

Hi, maybe I finded bug.
Not wokr rule  property "builtinGlobals" in rule "no-redeclare".
```js
/*eslint no-redeclare: ["error", { "builtinGlobals": true }]*/
/*eslint-env browser*/

var top = 0;
var Object = 0;
```

console message without error "no-redeclare"

```bash
4:1   error  Unexpected var, use let or const instead        no-var
  4:5   error  'top' is assigned a value but never used        no-unused-vars
  4:11  error  Number constants declarations must use 'const'  no-magic-numbers
  5:1   error  Unexpected var, use let or const instead        no-var
  5:5   error  'Object' is assigned a value but never used     no-unused-vars
  5:14  error  Number constants declarations must use 'const'  no-magic-numbers
```

**Comment by platinumazure** [comment]
Not 100% sure what's going on here, to be honest. Seems like a potential bug, but it might just be that your configuration needs to have a particular setting and we need to update the docs. Unfortunately, I can't tell immediately from the code.

**Comment by not-an-aardvark** [comment]
I can't reproduce this issue. `no-redeclare` is reporting an error with your code, as expected:

![](https://gyazo.com/4302c1d7f7b2a2e3358ba2af60365743.png)

**Comment by boggddan** [comment]
I turning offall rule in my configuration file. And testing with the clean project.
Maybe, linter on the site use other version and I run Eslint in console and vs code editor, error not popping.

My  `.eslint.js`
```js
module.exports = {
  env: {
    browser: true,
    es6: true,
    jquery: true,
    node: true
  },
  extends: 'eslint:recommended',
  parserOptions: {
    ecmaVersion: 2017,
    sourceType: 'script',
    ecmaFeatures: {
      experimentalObjectRestSpread: true
    },
  }
};
```
VSCode Editor:
![2017-06-27_09h09_01](https://user-images.githubusercontent.com/24915025/27573050-6459f8f6-5b18-11e7-8e67-9979f656cfb8.png)

Console:
![2017-06-27_09h12_12](https://user-images.githubusercontent.com/24915025/27573103-bf9e68c8-5b18-11e7-8c94-21d8eb5d1368.png)

**Comment by mysticatea** [comment]
Because you have `env: {node: true}` in your configuration.
The node environment has special function scope which encloses whole code. So variable declarations shadow globals but don't redeclare globals.

**Comment by platinumazure** [comment]
Per @mysticatea above, this is working as intended.

@boggddan I see that the `no-shadow` rule has [a similar `builtinGlobals` option](http://eslint.org/docs/rules/no-shadow#builtinglobals). I think that rule/option will do what you want. Please give it a shot, and if it doesn't work, please stop by our [chatroom](https://gitter.im/eslint/eslint). Thanks!

**Comment by boggddan** [comment]
@mysticatea, Thanks. It does work!
@platinumazure Could you make a note in the documentation about the fact that the `builtinGlobals: true` option. does not work when the parameter is switched on `env: {node: true}`. Thanks.

**Comment by platinumazure** [comment]
Good point @boggddan, we should add a note about module scope for Node and similar environments and recommend `no-shadow`'s equivalent option. Thanks!
